src/core/net/ip6_address.cpp

```diff
@@ -144,6 +144,26 @@ bool Address::IsIidReserved(void) const
     return IsSubnetRouterAnycast() || IsReservedSubnetAnycast() || IsAnycastRoutingLocator();
 }
 
+void Address::SetPrefix(const uint8_t *aPrefix, uint8_t aPrefixLength)
+{
+    uint8_t extraBits = (aPrefixLength % CHAR_BIT);
+
+    memcpy(mFields.m8, aPrefix, (aPrefixLength - extraBits) / CHAR_BIT);
```

Comment:
need to make sure `aPrefixLength <= 128`?

Reply:
Added an assert